docs/DeveloperGuide.adoc

```diff
@@ -763,20 +848,197 @@ Use case ends.
+4.  iContacts reads the file and merges the contacts from the received data file into the data file that iContacts is currently using
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. No file selected
++
+Use case ends.
+
+* 4a. iContacts finds duplicate contacts
++
+[none]
+** 4a1. iContacts shows an error message and doesn't merge the duplicate ones
++
+Use case resumes at step 1.
+
+[none]
+* 4b. iContacts unable to locate the path of file and open it
+** 4b1. iContacts shows an error message.
++
+Use case ends
+* *a. At any time user chooses to cancel importing contacts
++
+[none]
+** *a1. iContacts shows an error message.
++
+Use case ends.
+
+[discrete]
+=== Use case: UC4 - Sharing Contacts
+
+*Guarantees*
+
+[none]
+*  The sharing is completed with recipient successfully being able to get the shared contacts
+
+*MSS*
+
+1.  Sending User +++<u>Exports Contacts (UC2)</u>+++
+2.  Receiving User +++<u>Imports Contacts (UC3)</u>+++
+3.  Receiving User views the list of new contacts
+
++
+Use case ends.
+
+[discrete]
+=== Use case: UC5 - List contacts after applying filtered search
+
+*MSS*
+
+1.  User specifies the keywords for different link:#attributes[attributes] and requests the list
+2.  iContacts finds list of all people satisfying all attributes
+3.  iContacts displays list of all such contacts
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty
++
+Use case ends.
+
+[discrete]
+=== Use case: UC6 - Get directions to a contact's address
+
+*MSS*
+
+1.  User requests for the list of contacts
+2.  iContacts displays list of all such contacts
+3.  User requests for directions to a contact's address
+4.  iContacts shows directions from current location to contact's address in Google Maps using link:#GUI[GUI]
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. iContacts shows an error message.
++
+Use case resumes at step 2.
+
+[none]
+* 4a. Google Maps unable to find location of direction to address
++
+[none]
+** 4a1. iContacts shows an error message through GUI
++
+Use case ends.
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-{More to be added}
+.  A new user to a link:#CLI[CLI] environment and/or address book application should be able to learn how to install and use the application within 20 minutes with the help of the User Guide.
+.  Should be responsive and commands are executed within 2 seconds under normal load (1000 persons or less) and typical usage.
```

Comment:
what is typical usage? can you make it objective?

Reply:
The implication of typical usage is quite ambiguous so decided to omit the part